src/sensai/vector_model.py

```diff
@@ -434,73 +431,27 @@ def _predict(self, x: pd.DataFrame) -> pd.DataFrame:
         return self.convertClassProbabilitiesToPredictions(self._predictClassProbabilities(x))
 
 
-class RuleBasedModel(FittableModel, ABC):
-    """
-    Base class for models where the essential prediction logic is based on rules coded by humans
-    and thus does not require fitting. However, the input generation process may use mechanisms
-    such as feature generation or data transformation, which may require fitting.
-    """
+class RuleBasedRegressionModel(VectorRegressionModel):
```

Comment:
The structure is much simpler this way, thank you! What bugs me a bit is that rule based models now can have target transformer and output transformer. I can't really think of a case where it is ever useful. A hacky way to resolve this would be to override those methods as well, causing them to raise exceptions or log errors (the code would have to be duplicated too)

We could, of course, also take the approach to let the users think for themselves...

Reply:
I don't understand why that bugs you.
Having options is a good thing.

Reply:
More options lead to more room for errors (especially in design), that's why I mentioned it. I guess I worry too much about this anyway, probably no-one will even get the idea of using target transformer for rule based stuff

Reply:
It's true that some of these options don't make sense to use, but I think it would be highly unlikely for users to make use of them. So I really wouldn't worry.

Reply:
Hm, we could use the same trick as with the columnNameGetters here: raise an exception if the `withTarget/OutputTransformer` is called and the underlying model does not require fitting - as an indication that it doesn't make sense to use them